test/App/Fossa/YamlDepsSpec.hs

```diff
@@ -0,0 +1,64 @@
+module App.Fossa.YamlDepsSpec
+  ( spec,
+  )
+where
+
+import App.Fossa.YamlDeps
+  ( CustomDependency (CustomDependency),
+    ReferencedDependency (ReferencedDependency),
+    YamlDependencies (YamlDependencies),
+  )
+import Control.Effect.Exception (displayException)
+import Data.ByteString qualified as BS
+import Data.Yaml (decodeEither')
+import DepTypes (DepType (..))
+import Test.Hspec (Spec, describe, expectationFailure, it, runIO, shouldBe, Expectation, shouldContain)
+import Test.Hspec.Core.Spec (SpecM)
+
+uncurry3 :: (a -> b -> c -> d) -> (a, b, c) -> d
+uncurry3 f (a, b, c) = f a b c
+
+uncurry5 :: (a -> b -> c -> d -> e -> f) -> (a, b, c, d, e) -> f
+uncurry5 f (a, b, c, d, e) = f a b c d e
+
+getTestDataFile :: String -> SpecM a BS.ByteString
+getTestDataFile name = runIO . BS.readFile $ "test/App/Fossa/testdata/" <> name
+
+theWorks :: YamlDependencies
+theWorks = YamlDependencies references customs
+  where
+    references =
+      map
+        (uncurry3 ReferencedDependency)
```

Comment:
[nit] I think using `uncurry3`/`uncurry5` here doesn't help very much; might be better to inline the constructor in the list